src/ZiraLink.Api.Application/Tools/HttpTools.cs

```diff
@@ -48,5 +58,52 @@ public async Task<bool> PingDomain(string domainUrl)
 
             return true;
         }
+
+
+        public async Task<HttpClient> InitializeHttpClientAsync(CancellationToken cancellationToken)
+        {
+            var httpClient = _httpClientFactory.CreateClient(NamedHttpClients.Default);
+            var disco = await httpClient.GetDiscoveryDocumentAsync(_idsUri.ToString(), cancellationToken);
+            if (disco.IsError)
+                throw new ApplicationException("Failed to get discivery document");
+
+            var tokenResponse = await httpClient.RequestClientCredentialsTokenAsync(new ClientCredentialsTokenRequest
+            {
+                Address = disco.TokenEndpoint,
+
+                ClientId = "back",
+                ClientSecret = "secret",
+                Scope = "ziralink IdentityServerApi"
+            }, cancellationToken);
+
+            if (tokenResponse.IsError)
+                throw new ApplicationException("Failed to get token from identity server");
+
+            httpClient.SetBearerToken(tokenResponse.AccessToken);
+
+            return httpClient;
+        }
+
+        public async Task<T> CallIDSApis<T>(HttpClient httpClient, string relativeUri, object jsonObject, HttpMethod httpMethod, CancellationToken cancellationToken)
+        { 
+            var content = new StringContent(JsonSerializer.Serialize(jsonObject), Encoding.UTF8, "application/json");
+            var baseUri = _idsUri;
+            var uri = new Uri(baseUri, relativeUri);
+
+            HttpResponseMessage? response;
+
+            if (httpMethod.Equals(HttpMethod.Post)) response = await httpClient.PostAsync(uri.ToString(), content);
```

Comment:
This set of IFs makes the method way complex to understand, it is not the cleanest way to do that